File: javascript/bob/bob.js
export const hey = (message) => {
  if (/^\s*$/.test(message)) {
    return 'Fine. Be that way!';
  }
  if (/^[\sA-Z]+\?$/.test(message)) {
    return "Calm down, I know what I'm doing!";
  }
  if (/\?\s*$/.test(message)) {
    return 'Sure.';
  }
  if (/^[A-Z0-9\W]+$/.test(message) && /[A-Z]/.test(message)) {
    return 'Whoa, chill out!';
  }
  return 'Whatever.';
};

File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val isShouting = { input.matches(Regex(".*[A-Z]+.*")) && input.matches(Regex("^[^a-z]*$")) }
        return if (input.trimEnd().endsWith("?")) {
            if (isShouting()) {
                "Calm down, I know what I'm doing!"
            } else {
                "Sure."
            }
        } else if (isShouting()) {
            "Whoa, chill out!"
        } else if (input.matches(Regex("^\\s*$"))) {
            "Fine. Be that way!"
        } else {
            "Whatever."
        }
    }
}
